src/blockdata/transaction.rs

```diff
@@ -279,7 +290,16 @@ impl Transaction {
         // lock_time
         4;
         if inputs_with_witnesses == 0 {
-            non_input_size * 4 + input_weight
+            if self.input.is_empty() {
+                // If there are no inputs, then the transaction is serialized identically
+                // even after stripping witnesses (that is, even the segwit flag bytes are
+                // still there). Using the "length + 3 * stripped_length" formula, this
+                // means that the segwit flag bytes are considered to be non-witness bytes,
+                // and therefore have weight 8.
+                non_input_size * 4 + 8
```

Comment:
We're just gonna have to check this out in the fuzz test, it doesn't make sense to make get_weight not return the consensus-weight (which this isnt).

Reply:
Kk. My thinking was that there isn't really a "consensus weight" for an invalid transaction anyway.

Reply:
Hmm, thats a valid point, though I'd kinda prefer to at least get the same weight that Bitcoin Core will tell you a bogus transaction like this has, even if our serialization is different to avoid ambiguity (though, last I checked, our serialization would deserialize on Bitcoin Core RPC functions).

Reply:
Ok. I've fixed it, am fuzzing locally and will push when it finishes.